### PR TITLE
Blur Images

### LINKED ISSUE #1746: 🚀 [Feature] Blur explicit Image Content
## :rocket: Feature
The content of sensitive images NSFW (not safe for work) can be blurred so that it does not scare sensitive people.

### Design & Layout
Original Picture:
sensitive1

Blurred picture:
sensitive-2

Video: [Sensitive Filter](https://www.youtube.com/watch?v=9Oq6v7nG2XA)

### Additional context

It has to be defined, if just the author of an article should be able to mark it as sensitive or if any user should be able to do so. If so, it has to be decided, how the algorithm should work.

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
blur from disturbing images. 
you can blur an image in your contribution. 
everyone can view a blurred picture by pressing a button in the original. 

# Issues
fix #1746

## new contribution and blur
ezgif com-video-to-gif (1)

# contribution overview and unblur a blur image
ezgif com-video-to-gif (2)

#edit contribution
ezgif com-video-to-gif (3)

### CODE REVIEW

**Comment by ogerly** [comment]
ezgif com-video-to-gif

**Comment by DennisHack** [comment]
Possible to have a "disclaimer sign" (in red? SENSIBLE INHALTE/EXPLICIT CONTENT) on top of the blurred picture, when marked as sensitive?

File: backend/src/models/Post.js [hunk 1]
```diff
@@ -45,4 +45,5 @@ module.exports = {
     default: () => new Date().toISOString(),
   },
   language: { type: 'string', allow: [null] },
+  checkedBlur: { type: 'boolean', default: false },
```
Comment (mattwr18) [inline backend/src/models/Post.js]: I would rename `checkedBlur` to `blurImage`

File: webapp/components/ContributionForm/ContributionForm.spec.js [hunk 2]
```diff
@@ -241,6 +242,13 @@ describe('ContributionForm.vue', () => {
           expect(mocks.$apollo.mutate).toHaveBeenCalledTimes(1)
         })
 
+        describe('questionable images should be blurred', () => {
+          it('questionable images unset be blurred', async () => {
+            await wrapper.find('input[type="checkbox"]').trigger('click')
+            expect(wrapper.find('input[type="checkbox"]').exists()).toBe(true)
+          })
+        })
+
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.spec.js]: what is this even testing? you are finding an input of type checkbox, clicking on it and expecting that the element exists?? what am I missing?
It must exist, otherwise it would throw an error `Cannot trigger a click event on an empty Wrapper`, or something...
what happens when it's clicked? I guess it adds a class? so could we find the image and, `expect(img.classes()).toContain('blur-image')` or something similar?

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 3]
```diff
@@ -7,14 +7,56 @@
...
+        <div v-show="checkedBlur">
+          <ds-button
+            v-if="contribution"
+            class="bluricon-post"
+            icon="ban"
+            primary
+            @click.prevent="unBlur"
+          ></ds-button>
+          <img
+            v-if="contribution"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: can this duplicate `v-if` live in the parent of these two elements?

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 4]
```diff
@@ -7,14 +7,56 @@
...
+            :src="contribution.image | proxyApiUrl"
+            class="blurImgPreview"
+            @click.prevent="unBlur"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: same for the `click` event...

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 5]
```diff
@@ -7,14 +7,56 @@
...
+            v-if="contribution"
+            class="bluricon-post"
+            icon="ban"
+            primary
+            @click.prevent="unBlur"
+          ></ds-button>
+          <img
+            v-if="contribution"
+            :src="contribution.image | proxyApiUrl"
+            class="blurImgPreview"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: From what I understand it's good to, `Use Hyphen Delimited Strings` for `CSS` names, no @alina-beck??
Here for reference https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 6]
```diff
@@ -7,14 +7,56 @@
...
+        </div>
+      </ds-text>
+      <div style="clear: both" />
       <ds-card>
+        <ds-text align="right">
+          <label for="blur_img">{{ $t('contribution.shockingPicture') }}</label>
+          <input
+            name="checkedBlur"
+            type="checkbox"
+            id="blur_img"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: see :point_up_2:

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 7]
```diff
@@ -183,6 +239,28 @@ export default {
...
+        this.checkedBlur = false
+        this.form.checkbox = false
+      } else {
+        if (this.elem != null) {
+          this.elem.classList.add('img-blur-in')
+        }
+        this.checkedBlur = true
+        this.form.checkbox = true
+      }
+    },
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]: I think we can improve this! there is an empty else after `line 250`... 
there are nested `if` inside `else` clauses, which could be refactored to use `else if` statements, no?
Also, it's a bit complicated to follow what's going on here... what are you trying to achieve in `checkedChange`?
I see it is called when `unBlur` is called, and when the component is `mounted`

File: webapp/components/TeaserImage/TeaserImage.vue [hunk 8]
```diff
@@ -119,6 +120,7 @@ export default {
       }, 'image/jpeg')
     },
     setupPreview(canvas) {
+      // console.log(' new Image setupPreview')
```
Comment (mattwr18) [inline webapp/components/TeaserImage/TeaserImage.vue]:
```suggestion

```

File: webapp/components/TeaserImage/TeaserImage.vue [hunk 9]
```diff
@@ -103,6 +103,7 @@ export default {
       if (contributionImage) contributionImage.remove()
     },
     initCropper() {
+      // console.log(' new Image initCropper')
```
Comment (mattwr18) [inline webapp/components/TeaserImage/TeaserImage.vue]:
```suggestion

```

File: webapp/locales/en.json [hunk 10]
```diff
@@ -644,7 +644,9 @@
     },
     "teaserImage": {
       "cropperConfirm": "Confirm"
-    }
+    },
+    "shockingPicture" : "This image may be inappropriate for some people.",
+    "shockingPicture-text" : "Contribution picture is blurred"
```
Comment (mattwr18) [inline webapp/locales/en.json]:
```suggestion
    "inappropriatePicture-text" : "Contribution picture is blurred"
```

File: webapp/locales/en.json [hunk 11]
```diff
@@ -644,7 +644,9 @@
     },
     "teaserImage": {
       "cropperConfirm": "Confirm"
-    }
+    },
+    "shockingPicture" : "This image may be inappropriate for some people.",
```
Comment (mattwr18) [inline webapp/locales/en.json]:
```suggestion
    "inappropriatePicture" : "This image may be inappropriate for some people.",
```

File: webapp/locales/de.json [hunk 12]
```diff
@@ -429,8 +430,9 @@
         },
         "teaserImage": {
             "cropperConfirm": "Bestätigen"
-        },
-        "languageSelectText": "Sprache wählen"
+        },        
+        "shockingPicture" : "Dieses Bild kann für einige Menschen unangemessen sein.",
+        "shockingPicture-text" : "Beitragsbild wird Verschwommen dargestellt"
```
Comment (mattwr18) [inline webapp/locales/de.json]:
```suggestion
       "inappropriatePicture-text" : "Beitragsbild wird Verschwommen dargestellt"
```

File: webapp/locales/de.json [hunk 13]
```diff
@@ -429,8 +430,9 @@
         },
         "teaserImage": {
             "cropperConfirm": "Bestätigen"
-        },
-        "languageSelectText": "Sprache wählen"
+        },        
+        "shockingPicture" : "Dieses Bild kann für einige Menschen unangemessen sein.",
```
Comment (mattwr18) [inline webapp/locales/de.json]:
```suggestion
        "inappropriatePicture" : "Dieses Bild kann für einige Menschen unangemessen sein.",
```

File: webapp/pages/post/_id/_slug/index.vue [hunk 14]
```diff
@@ -150,6 +177,18 @@ export default {
     },
   },
   methods: {
+    unBlur() {
+      if (this.post.checkedBlur) {
+        this.post.checkedBlur = false
+        this.blur = true
+      }
+    },
+    Blur() {
```
Comment (mattwr18) [inline webapp/pages/post/_id/_slug/index.vue]:
```suggestion
    blur() {
```

File: webapp/pages/post/_id/_slug/index.vue [hunk 15]
```diff
@@ -4,9 +4,35 @@
...
+      <ds-text align="right" class="blurBox">
+        <ds-button
+          v-show="post.checkedBlur"
+          class="bluricon-post"
+          icon="eye"
+          primary
+          @click.prevent="unBlur"
+        ></ds-button>
+        <ds-button
+          v-show="blur"
```
Comment (mattwr18) [inline webapp/pages/post/_id/_slug/index.vue]:
```suggestion
          v-show="blurred"
```

File: webapp/pages/post/_id/_slug/index.vue [hunk 16]
```diff
@@ -125,6 +151,7 @@ export default {
       ready: false,
       title: 'loading',
       showNewCommentForm: true,
+      blur: false,
```
Comment (mattwr18) [inline webapp/pages/post/_id/_slug/index.vue]:
```suggestion
      blurred: false,
```

File: webapp/pages/post/_id/_slug/index.vue [hunk 17]
```diff
@@ -150,6 +177,18 @@ export default {
...
+    unBlur() {
+      if (this.post.checkedBlur) {
+        this.post.checkedBlur = false
+        this.blur = true
+      }
+    },
+    Blur() {
+      if (!this.post.checkedBlur) {
+        this.post.checkedBlur = true
+        this.blur = false
```
Comment (mattwr18) [inline webapp/pages/post/_id/_slug/index.vue]:
```suggestion
        this.blurred = false
```

File: webapp/pages/post/_id/_slug/index.vue [hunk 18]
```diff
@@ -206,8 +245,38 @@ export default {
...
+  width: 100px;
+}
+
+.bluricon-post {
+  font-size: xx-large;
+  /* padding: 13px; */
+  /* position: absolute; */
+  background-color: green;
+  /* top: 131px; */
+}
```
Comment (mattwr18) [inline webapp/pages/post/_id/_slug/index.vue]: remove uncommented out CSS?

File: webapp/locales/en.json (hunk 10 above)
Comment (roschaefer) [reply webapp/locales/en.json]:
```suggestion
    "inappropriatePictureText" : "Contribution picture is blurred"
```

File: backend/src/models/Post.js (hunk 1 above)
Comment (ogerly) [reply backend/src/models/Post.js]: ok, i changed the name in the whole system

File: webapp/pages/post/_id/_slug/index.vue (hunk 14 above)
Comment (ogerly) [reply webapp/pages/post/_id/_slug/index.vue]: ok renamed

File: webapp/components/ContributionForm/ContributionForm.spec.js [hunk 19]
```diff
@@ -241,6 +242,17 @@ describe('ContributionForm.vue', () => {
           expect(mocks.$apollo.mutate).toHaveBeenCalledTimes(1)
         })
 
+        describe('questionable images should be blurred', () => {
+          it('questionable images unset be blurred', async () => {
+            expectedParams.variables.imageUpload = imageUpload
+            wrapper.find(TeaserImage).vm.$emit('addTeaserImage', imageUpload)
+            expect(wrapper.find('.images-set-blur').exists()).toBe(false)
+            await wrapper.find('input[type="checkbox"]').trigger('click')
+            expect(wrapper.find('input[type="checkbox"]').exists()).toBe(true)
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.spec.js]:
```suggestion
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 20]
```diff
@@ -7,14 +7,57 @@
     @submit="submit"
   >
     <template slot-scope="{ errors }">
-      <hc-teaser-image :contribution="contribution" @addTeaserImage="addTeaserImage">
+      <hc-teaser-image
+        :contribution="contribution"
+        @addTeaserImage="addTeaserImage"
+        :class="{ 'images-set-blur': checkedBlur }"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
        :class="{ 'images-set-blur': blurImage }"
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 21]
```diff
@@ -7,14 +7,57 @@
...
+      >
         <img
           v-if="contribution"
           class="contribution-image"
           :src="contribution.image | proxyApiUrl"
         />
       </hc-teaser-image>
+      <ds-text align="right" class="blurBox">
+        <div v-show="checkedBlur">
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
        <div v-show="blurImage">
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 22]
```diff
@@ -7,14 +7,57 @@
...
+            v-if="contribution"
+            class="bluricon-post"
+            icon="ban"
+            primary
+            @click.prevent="unBlur"
+          ></ds-button>
+          <img
+            v-if="contribution"
+            :src="contribution.image | proxyApiUrl"
+            class="blurImgPreview"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
            class="blur-img-preview"
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 23]
```diff
@@ -7,14 +7,57 @@
...
+          />
+        </div>
+      </ds-text>
+      <div style="clear: both" />
       <ds-card>
+        <ds-text align="right">
+          <label for="blur_img">{{ $t('contribution.shockingPicture') }}</label>
+          <input
+            name="checkbox"
+            class="blurImageCheckbox"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
            class="blur-image-checkbox"
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 24]
```diff
@@ -7,14 +7,57 @@
...
+            type="checkbox"
+            id="blur_img"
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
            id="blur-img"
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 25]
```diff
@@ -172,6 +220,14 @@ export default {
       users: [],
       contentMin: 3,
       hashtags: [],
+      elem: null,
+      checkedBlur: false,
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
      blurImage: false,
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 26]
```diff
@@ -172,6 +220,14 @@ export default {
...
+    }
+  },
+  created() {},
+  mounted() {
+    if (this.contribution && this.contribution.blurImage === true) {
+      this.checkedBlur = true
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
      this.blurImage = true
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 27]
```diff
@@ -183,6 +239,11 @@ export default {
     }),
   },
   methods: {
+    unBlur() {
+      if (this.checkedBlur) {
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
      if (this.blurImage) {
```

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 28]
```diff
@@ -183,6 +239,11 @@ export default {
...
+        this.checkedBlur = false
```
Comment (mattwr18) [inline webapp/components/ContributionForm/ContributionForm.vue]:
```suggestion
        this.blurImage = false
```

File: webapp/components/ContributionForm/ContributionForm.spec.js (hunk 2 above)
Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.spec.js]: changed

File: backend/src/models/Post.js [hunk 29]
```diff
@@ -39,5 +39,6 @@ module.exports = {
     default: () => new Date().toISOString(),
   },
   language: { type: 'string', allow: [null] },
+  blurImage: { type: 'boolean', default: false },
```
Comment (mattwr18) [inline backend/src/models/Post.js]:
```suggestion
  imageBlurred: { type: 'boolean', default: false },
  imageBlurredByModerator: { type: 'boolean', default: false },
```

Comment (mattwr18) [reply backend/src/models/Post.js]: talking about it, it would be nice to keep the image properties in the database start with `image`

File: webapp/components/ContributionForm/ContributionForm.vue (hunk 25 above)
Comment (mattwr18) [reply webapp/components/ContributionForm/ContributionForm.vue]: then we can keep this `blurImage` and it's not as confusing
`blur` is a verb, so the user is doing and action, so the naming fits better

File: backend/src/models/Post.js (hunk 29 above)
Comment (ogerly) [reply backend/src/models/Post.js]: ok

**Comment by Tirokk** [comment]
@ogerly I tried the blurring out. Looks really great so far. 😍

Some questions:

- As far I see some or even all of @mattwr18 change requests are not implemented. As an example `blurImage` should be named `imageBlurred` in the database. And so on …
- The blur extends over the border or area of the post or post card on the post page and in the newsfeed. Is that supposed to be this way?
- If the image is blurred there is the small thumbnail in the down right corner …
  - My eyes are not so bad and I ask myself: If a image is offending or be inappropriate for some people even this small thumbnail would do this to me in my eyes.
    Did you thought about this?

File: webapp/components/ContributionForm/ContributionForm.vue [hunk 30]
```diff
@@ -177,6 +200,13 @@ export default {
       users: [],
       contentMin: 3,
       hashtags: [],
+      elem: null,
+      blurImage: false,
+    }
+  },
+  mounted() {
+    if (this.contribution && this.contribution.blurImage === true) {
+      this.blurImage = true
```
Comment (alina-beck) [inline webapp/components/ContributionForm/ContributionForm.vue]: @ogerly please check how all the other form values are initiated – I know it's a bit confusing but I think we should stick to the existing behavior.

This would also mean we don't need the `mounted()` block anymore, which is always good, and we don't need to double the value for blurring images as `checkbox` and `blurImage` – it could just be one `blurImage` value that responds to the checkbox input!

Comment (ogerly) [reply webapp/components/ContributionForm/ContributionForm.vue]: mounted and put everything on blurImage. 
i hope i understood that correctly :)

**Comment by ogerly** [comment]
> @ogerly habe ich versucht , die Unschärfe aus. Sieht soweit echt toll aus.
> 
> Einige Fragen:
> 
> * Soweit ich sehe, sind einige oder sogar alle @ mattwr18- Änderungsanforderungen nicht implementiert. Als Beispiel `blurImage`soll `imageBlurred`in der Datenbank genannt werden. Und so weiter …
> * Die Unschärfe erstreckt sich über den Rand oder Bereich des Postens oder der Postkarte auf der Postseite und im Newsfeed. Soll das so sein?
> * Wenn das Bild unscharf ist, befindet sich das kleine Vorschaubild in der unteren rechten Ecke.
>   
>   * Meine Augen sind nicht so schlecht und ich frage mich: Wenn ein Bild für manche Menschen unangemessen oder unangebracht ist, würde mir sogar dieses kleine Vorschaubild dies in meinen Augen antun. 
>     Hast du darüber nachgedacht?

alina cleaned the edges. 
i'm still changing the name for the database. that should be in the new PR but that doesn't make any sense, i can only see that now. 
and the thumbnail is really diskutable. but it would be a mini change if you want it different.

**Comment by ogerly** [comment]
@Tirokk alina cleaned the edges. 
i'm still changing the name for the database. that should be in the new PR but that doesn't make any sense, i can only see that now. 
and the thumbnail is really diskutable. but it would be a mini change if you want it different.

**Comment by Tirokk** [comment]
Are you still working on changing the property names of the database?
Seems not to be yet in the code …

With the thumbnail I would ask @datenbrei .

I test it and will have a watch on the blur, what @alina-beck did, after breakfast in kind of an hour.

File: backend/src/models/Post.js (hunk 29 above)
Comment (ogerly) [reply backend/src/models/Post.js]: I changed the database name from "blurImage" to "imageBlurred".

**Comment by ogerly** [comment]
@Tirokk I changed the database name from "blurImage" to "imageBlurred".

**Comment by ogerly** [comment]
@Tirokk @roschaefer @mattwr18 

-  blurImageCheckbox css is delete
- all console commands are deleted
-   I have _shockingPicture_  renamed to  _inappropriatePicture_ and _inappropriatePictureText_

**Comment by ogerly** [comment]
ezgif com-video-to-gif (5)

File: webapp/components/PostCard/PostCard.vue [hunk 31]
```diff
@@ -141,25 +146,37 @@ export default {
       this.$emit('unpinPost', post)
     },
   },
-}
-</script>
+  mounted() {
+    const width = this.$el.offsetWidth
+    const height = Math.min(width / this.post.imageAspectRatio, 2000)
+    const imageElement = this.$el.querySelector('.ds-card-imag')
```
Comment (mattwr18) [inline webapp/components/PostCard/PostCard.vue]:
```suggestion
    const imageElement = this.$el.querySelector('.ds-card-image')
```

Comment (mattwr18) [reply webapp/components/PostCard/PostCard.vue]: Also, I'm concerned about this because we have not introduced `imageAspectRatio` for the overwhelming majority of `Post`... new `Posts` add their `imageAspectRatio`, but we would need to convert more than 18,000 `Post` where this is null... so what happens in that case?

Comment (ogerly) [reply webapp/components/PostCard/PostCard.vue]: @mattwr18  i have change .ds-card-imag(e)

Comment (alina-beck) [reply webapp/components/PostCard/PostCard.vue]: I don't know how this code made its way into this PR – maybe from merging in `master` while we had it in there? – but it's breaking the layout. This is what happens to images without aspect ratio:
Screenshot 2019-12-16 at 15 30 41

Also, weirdly, when I edit a `Post` – even without changing the picture or `blurred` status the post loses its `imageAspectRatio`. So unfortunately there's more cleaning up to do! 😞

Comment (ogerly) [reply webapp/components/PostCard/PostCard.vue]: cheetah-rain

File: webapp/locales/de.json [hunk 32]
```diff
@@ -484,8 +485,9 @@
         },
         "teaserImage": {
             "cropperConfirm": "Bestätigen"
-        },
-        "languageSelectText": "Sprache wählen"
+        },        
+        "inappropriatePicture" : "Dieses Bild kann für einige Menschen unangemessen sein.",
+        "inappropriatePictureText" : "Beitragsbild wird Verschwommen dargestellt"
```
Comment (alina-beck) [inline webapp/locales/de.json]: one last, tiny change – since this is the link text that leads to the FAQ, I would suggest something like:
```suggestion
        "inappropriatePictureText" : "Wann soll ein Foto versteckt werden"
```

File: webapp/locales/en.json [hunk 33]
```diff
@@ -702,7 +702,9 @@
     },
     "teaserImage": {
       "cropperConfirm": "Confirm"
-    }
+    },
+    "inappropriatePicture" : "This image may be inappropriate for some people.",
+    "inappropriatePictureText" : "Contribution picture is blurred"
```
Comment (alina-beck) [inline webapp/locales/en.json]: and in English:
```suggestion
    "inappropriatePictureText" : "When should my picture be blurred"
```

File: .vscode/settings.json [hunk 34]
```diff
@@ -8,5 +8,8 @@
     }
   ],
   "editor.formatOnSave": false,
-  "eslint.autoFixOnSave": true
+  "eslint.autoFixOnSave": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
```
Comment (roschaefer) [inline .vscode/settings.json]: Unrelated changes?

File: backend/src/seed/seed-db.js [hunk 35]
```diff
@@ -360,72 +361,83 @@ const languages = ['de', 'en', 'es', 'fr', 'it', 'pt', 'pl']
...
         categoryIds: ['cat14'],
+        imageBlurred: false,
         imageAspectRatio: 300 / 450,
       }),
       factory.create('Post', {
         author: huey,
         id: 'p15',
         language: sample(languages),
         categoryIds: ['cat15'],
+        imageBlurred: false,
```
Comment (roschaefer) [inline backend/src/seed/seed-db.js]: Can't we make `imageBlurred: false` a default in the post factory?

File: webapp/locales/de.json (hunk 32 above)
Comment (ogerly) [reply webapp/locales/de.json]: ok

File: webapp/locales/en.json (hunk 33 above)
Comment (ogerly) [reply webapp/locales/en.json]: ok

File: .vscode/settings.json (hunk 34 above)
Comment (ogerly) [reply .vscode/settings.json]: that came with a git pull origin 174 ....

Comment (alina-beck) [reply .vscode/settings.json]: I had the same thing – it's an `ESLint` plugin change that came with the latest `VSCode` update. The question is: Do we want this setting activated for all developers? Or should we go the other direction and add `.vscode/settings.json` to `.gitignore`?

File: webapp/locales/de.json [hunk 36]
```diff
@@ -452,6 +452,7 @@
         "filterALL": "Alle Beiträge anzeigen",
         "success": "Gespeichert!",
         "languageSelectLabel": "Sprache",
```
Comment (Tirokk) [inline webapp/locales/de.json]: In the english version it is named:
`"languageSelectLabel": "Language of your contribution“`

In german that would be:
"languageSelectLabel": „Sprache deines Beitrags“

I would like to change this, because a lot of people seem to think this is as well the language of the web app and not of the post.
But anyways. This doesn’t hold me back to approve and we can even do it in an other PR.

Comment (ogerly) [reply webapp/locales/de.json]: https://github.com/Human-Connection/Human-Connection/issues/2583
